src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodDriver : GpioDriver
+    {
+        private const string GpioGet = "gpioget";
+        private const string GpioChip0 = "gpiochip0 ";
+        private const string GpioSet = "gpioset";
+        public int _waitMilliSeconds { get; set;}
+        private Dictionary<int, PinMode> _pinNumperToPinModes;
+        private Dictionary<int, Process> _pinNumperToProcess;
+        protected internal override int PinCount => throw new PlatformNotSupportedException("This driver is generic so it can not enumerate how many pins are available.");
+
+        public LibgpiodDriver() {
+            _pinNumperToPinModes = new Dictionary<int, PinMode>();
+            _pinNumperToProcess = new Dictionary<int, Process>();
+            _waitMilliSeconds = (int)TimeSpan.FromSeconds(1).TotalMilliseconds;
+        }
+        protected internal override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventType, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void ClosePin(int pinNumber)
+        {
+            // do nothing_pinNumperToProcess
+        }
+
+        protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => throw new PlatformNotSupportedException("This driver is generic so it can not perform conversions between pin numbering schemes.");
+
+        protected internal override PinMode GetPinMode(int pinNumber)
+        {
+            if (_pinNumperToPinModes.ContainsKey(pinNumber))
+             {
+                return _pinNumperToPinModes[pinNumber];
+            }
+            else
+            {
+                throw new InvalidOperationException("There was an attempt to get a mode to a pin that is not yet open.");
```

Comment:
No need to do this. In theory this should already be taken care of by the controller.

Reply:
That is true, we have that in the unix driver as well, that said, I think we shouldn't be hitting that code path because here is our One and Only callsite: https://github.com/dotnet/iot/blob/master/src/System.Device.Gpio/System/Device/Gpio/GpioController.cs#L127-L135

As you can see we are already guarding against this.

> The default value is input maybe return that?

No. I think we should have the controller throw an exception in this case.

Reply:
Yeah got it, controller throwing this exception so no need throw it here, but in case we are not throwing an exception we need to return some default value at this point